torch/_subclasses/meta_utils.py

```diff
@@ -269,6 +269,99 @@ def empty_create(inner_t, inner_src, symbolic_context=symbolic_context):
+            # to construct the (symbolic) sizes of the wrapper tensor.
+            return transform_subclass(
+                t,
+                lambda attr, inner_t: callback(
+                    lambda: empty_create(
+                        inner_t,
+                        AttrSource(source, attr),
+                        symbolic_context=(
+                            None
+                            if symbolic_context is None
+                            else symbolic_context.inner_contexts[attr]
+                        ),
+                    )
+                ),
+                outer_size=outer_size,
+                outer_stride=outer_stride,
+            )
+
+        # Returns a fake-ified version of t with the same view relationship wrt the given
+        # fake base. For views involving subclasses, this performs view replay, while
+        # dense -> dense views can be simplified to an as_strided() call. Note that view
+        # replay involves handling any closed-over state present in the view func:
+        #   * Closed-over tensors are fake-ified
+        #   * Closed-over SymInts are made symbolic
+        def view_from_base(base, t, source=source, shape_env=shape_env):
+            # fake-ify t naively; view relationship isn't correct yet
+            (sizes, strides, storage_offset) = sym_sizes_strides_storage_offset(
+                t, source
+            )
+            if not is_traceable_wrapper_subclass(
+                t
+            ) and not is_traceable_wrapper_subclass(base):
+                # Dense -> Dense view case uses as_strided() to construct view relationship.
+                # TODO: Change this logic to use view replay for consistency?
+                # It's likely there is no view func available.
+                return base.as_strided(sizes, strides, storage_offset)
+
+            def symint_visitor_fn(s):
+                if shape_env is None:
+                    return s
+
+                from torch._dynamo.source import ConstantSource
+
+                # NB: The source is meaningless; the symbol here is expected to be simplified out.
+                sym_source = ConstantSource("view_func")
+                val = int(s)
+                symbol = shape_env.create_symbol(val, sym_source)
+                return shape_env.create_symintnode(symbol, hint=val, source=sym_source)
+
+            # Assume the only closed-over tensors encountered will be subclass inner tensors.
+            real_to_fake_mapping = {}
+            if is_traceable_wrapper_subclass(t):
+                fake_t = empty_create_subclass(
+                    t, outer_size=sizes, outer_stride=strides
+                )
+                attrs, _ = fake_t.__tensor_flatten__()
+                for attr in attrs:
+                    real_to_fake_mapping[getattr(t, attr)] = getattr(fake_t, attr)
+
+            def tensor_visitor_fn(t):
+                return real_to_fake_mapping.get(t, t)
```

Comment:
Why is it OK to return the tensor t here directly if you don't find it in the map? I would have expected either (1) everything is guaranteed to be in the mapping or (2) you fakeify if it's not in the mapping (but it seems like you are trying to do that above).

Reply:
Addressed this via `EphemeralSource` - now any tensors not in the mapping are fully-dynamically fake-ified with `EphemeralSource`s